src/WebJobs.Extensions.DurableTask/Listener/TaskActivityShim.cs

```diff
@@ -61,6 +61,13 @@ public override async Task<string> RunAsync(TaskContext context, string rawInput
                 // without the outer FunctionInvocationException.
                 Exception exceptionToReport = StripFunctionInvocationException(result.Exception);
 
+                // If the exception looks like an RPC exception, wrap it in an OutOfProcessException
+                // to strip unhelpful information from message.
+                if (OutOfProcExceptionHelpers.IsRpcException(exceptionToReport))
```

Comment:
We can name the method differently. The reason I named it `IsRpcException` is because it is a best effort to catch the exception type `RpcException`, which is thrown by the Functions runtime. Eventually, this check could be replaced with a check of `exceptionToReport is RpcException`.

That being said, that is kind of an implementation detail, and I can definitely see the merit in renaming this `IsOutOfProcException` to hide that detail.